app/controllers/api/projects.php

```diff
@@ -121,40 +123,53 @@
+
+        $wg = new WaitGroup();
+
         foreach ($collections as $key => $collection) {
-            if (($collection['$collection'] ?? '') !== Database::METADATA) {
-                continue;
-            }
-            $attributes = [];
-            $indexes = [];
-
-            foreach ($collection['attributes'] as $attribute) {
-                $attributes[] = new Document([
-                    '$id' => $attribute['$id'],
-                    'type' => $attribute['type'],
-                    'size' => $attribute['size'],
-                    'required' => $attribute['required'],
-                    'signed' => $attribute['signed'],
-                    'array' => $attribute['array'],
-                    'filters' => $attribute['filters'],
-                    'default' => $attribute['default'] ?? null,
-                    'format' => $attribute['format'] ?? ''
-                ]);
-            }
+            go(function () use ($collection, $dbForProject, $key, $wg) {
+                $wg->add();
 
-            foreach ($collection['indexes'] as $index) {
-                $indexes[] = new Document([
-                    '$id' => $index['$id'],
-                    'type' => $index['type'],
-                    'attributes' => $index['attributes'],
-                    'lengths' => $index['lengths'],
-                    'orders' => $index['orders'],
-                ]);
-            }
+                if (($collection['$collection'] ?? '') !== Database::METADATA) {
+                    return;
+                }
+                $attributes = [];
+                $indexes = [];
+
+                foreach ($collection['attributes'] as $attribute) {
+                    $attributes[] = new Document([
+                        '$id' => $attribute['$id'],
+                        'type' => $attribute['type'],
+                        'size' => $attribute['size'],
+                        'required' => $attribute['required'],
+                        'signed' => $attribute['signed'],
+                        'array' => $attribute['array'],
+                        'filters' => $attribute['filters'],
+                        'default' => $attribute['default'] ?? null,
+                        'format' => $attribute['format'] ?? ''
+                    ]);
+                }
+
+                foreach ($collection['indexes'] as $index) {
+                    $indexes[] = new Document([
+                        '$id' => $index['$id'],
+                        'type' => $index['type'],
+                        'attributes' => $index['attributes'],
+                        'lengths' => $index['lengths'],
+                        'orders' => $index['orders'],
+                    ]);
+                }
 
-            $dbForProject->createCollection($key, $attributes, $indexes);
+                $dbForProject->createCollection($key, $attributes, $indexes);
+                $wg->done();
+            });
         }
 
+        $wg->wait(5);
```

Comment:
Is this value in second or ms?

Reply:
Seconds, That call doesn't actually wait in the usual way. It waits for all the coroutines to run; that 5 seconds is the maximum time, it will wait.